### PR TITLE
AIMD approach for tcp friendly window

### LINKED ISSUE #20: Replace modelled TCP Reno window approach with AIMD emulation
Yuchung Cheng [wrote](https://mailarchive.ietf.org/arch/msg/tcpm/KUQtS5nnRA1TaBzF2M2tnmTryGQ):
>I'd recommend replacing the modelled TCP Reno window approach in
section 4.2 with an AIMD emulation (Linux's approach).
>
>In our experience, TCP-friendly regions are the predominant mode of
(Linux) Cubic for any regular Internet connection. IOW Cubic is often
"Reno" unless the loss rate is abysmal. The modelled approach is based
on a simple bulk transfer where modern network applications are mostly
structured traffic (burst, idle, repeat). Under such traffic
structures the model has two issues:
>
>The model assumes cwnd overshoot causes losses that are repaired in
one round of fast recovery. In reality, the losses are often due to
bursts to short messages, causing more rounds and even timeouts to
repair. So the overall loss rate "p" tends to be higher than the ideal
model, causing the model to underestimate the window (hence runs in a
more conservative Reno). Instead Linux's approach is to simply emulate
Reno AIMD based on the number of packets per ACK. This also avoids
square-root operation.

### PR DESCRIPTION
**Pull request by goelvidhi**
Closes #20 .

Will add the change log once the pending PR #3 is merged to avoid conflict.

### CODE REVIEW

**Review by lisongxu — APPROVED** [review]
Thank you, Vidhi!

**Review by sangtaeha — APPROVED** [review]
This looks good. Thanks!

**Comment by goelvidhi** [comment]
@larseggert could you please review this PR?

File: draft-eggert-tcpm-rfc8312bis.md [hunk 1]
```diff
@@ -424,13 +424,15 @@ alpha_aimd=3*(1-beta_cubic)/(1+beta_cubic) and beta_aimd=beta_cubic,
...
+
+W_est is set equal to cwnd at the start of the congestion avoidance
+stage. After that, on every ACK, W_est is updated using Eq. 4.
 
 ~~~
-    W_est(t) = W_max * beta_cubic +
-               [3 * (1 - beta_cubic) / (1 + beta_cubic)] *
-               (t / RTT)                              (Eq. 4)
+    W_est += [3 * (1 - beta_cubic) / (1 + beta_cubic)] *
+              (segments_acked / cwnd)                 (Eq. 4)
```
Comment (larseggert) [inline draft-eggert-tcpm-rfc8312bis.md:435]: We're no using `+=` or similar anywhere in the formulas yet. So maybe use `W_est = W_est + ...` for consistency?

**Review by larseggert — CHANGES_REQUESTED** [review]
Also add a line to the changelog, please.

**Comment by larseggert** [comment]
FYI, I am technically on vacation this week, so I may be slow to respond.

File: draft-eggert-tcpm-rfc8312bis.md (hunk 1 above)
Comment (goelvidhi) [reply draft-eggert-tcpm-rfc8312bis.md]: sounds good. Done.

**Comment by goelvidhi** [comment]
> Also add a line to the changelog, please.

Done.

**Pull request merged**